src/beast/beast/http/basic_parser.h

```diff
@@ -90,20 +107,23 @@ class basic_parser
 public:
     using error_code = boost::system::error_code;
 
-    virtual
-    ~basic_parser() = default;
```

Comment:
Why? the class is not meant to be derived from. You could argue for `final`. But not `virtual`.